From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Piotr Filiciak <piotr@filiciak.pl>
Date: Fri, 8 May 2026 18:35:16 +0200
Subject: [PATCH] gs_free fix

---
 properties/nm-openvpn-editor.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/properties/nm-openvpn-editor.c b/properties/nm-openvpn-editor.c
index 5f400a9..127a04b 100644
--- a/properties/nm-openvpn-editor.c
+++ b/properties/nm-openvpn-editor.c
@@ -264,7 +264,7 @@ pkcs11_populate_ids_for_provider (GtkComboBoxText *id_combo, const char *provide
 	        PKCS11H_ENUM_METHOD_CACHE_EXIST, NULL,
 	        PKCS11H_PROMPT_MASK_ALLOW_ALL, NULL, &certs) == CKR_OK) {
 		for (cur = certs; cur != NULL; cur = cur->next) {
-			char *ser = NULL;
+			gs_free char *ser = NULL;
 			size_t ser_len = 0;
 
 			if (pkcs11h_certificate_serializeCertificateId (
@@ -274,7 +274,6 @@ pkcs11_populate_ids_for_provider (GtkComboBoxText *id_combo, const char *provide
 			if (pkcs11h_certificate_serializeCertificateId (
 			        ser, &ser_len, cur->certificate_id) == CKR_OK)
 				gtk_combo_box_text_append_text (id_combo, ser);
-			g_free (ser);
 		}
 		pkcs11h_certificate_freeCertificateIdList (certs);
 	}
